platform/core/src/classes/MetadataProvider.js

```diff
@@ -443,17 +443,7 @@ class MetadataProvider {
       };
     }
 
-    // Maybe its a non-standard imageId
-    // check if the imageId starts with http:// or https:// using regex
-    // Todo: handle non http imageIds
-    let imageURI;
-    const urlRegex = /^(http|https):\/\//;
-    if (urlRegex.test(imageId)) {
-      imageURI = imageId;
-    } else {
-      imageURI = imageIdToURI(imageId);
-    }
-
+    const imageURI = imageIdToURI(imageId);
```

Comment:
I'm pretty sure there was a reson we had the regex check, why we don't need it anymore?

Reply:
because this check introduced a bug by not passing the imageId to the `imageIdToURI` function in the http/https case. the function removes the protocol from the imageId, which is usually wadors:, wadourl:..., and maintains a valid URI, so while the concept of having the regex to verify if it is a standard URL then it doesn't need to "convert" makes sense, that check should be in that method `imageIdToURI` not here. 

For the code here, the imageURI is being used as a lookup in a map for the UIDs, which is maintained in this MetadataProvider class and uses the same approach for adding UIDs (calling `imageIdToURI`) directly without the regex check.

so the options we have:
1. Remove the check here, it should have 0 drawbacks and least effort
2. Do the regex test everywhere needed in the MetadataProvider - not good option at all as you are getting 0 value from that vs. option 1 and more code + effort
3. Update the method `imageIdToURI` to test for the regex. Utlimately, this should be our NorthStar. However, this is a shared method and used several other areas in the code and they all need to be tested if we change the behavior. Something that I don't intend doing in this PR unless there is a reason we need to.

Reply:
I don't understand, hmm. Isn't it the fact that if imageId is `http/https...` then it is already imageURI and no need to be converted?